worker_server.py

```diff
@@ -32,7 +36,7 @@ def SubmitTask(self, request, context):
         return api_pb2.TaskResponse(future=api_pb2.Future(resultLocation=future.resultLocation, hostName=future.hostName, port=future.port))
 
 def serve():
-    server = grpc.server(futures.ThreadPoolExecutor())
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers = MAX_WORKERS_COUNT))
```

Comment:
Threadpool here is for gRPC to accept connections concurrently in multiple threads. I think a more appropriate constraint should be done in `submit_task` method in worker.py where we execute the task in a thread (spawning of threads here should be capped based on user input)

Reply:
I would like to discuss this more - lets get this PR in and we can then modify after our discussion tomorrow.

Reply:
Updated